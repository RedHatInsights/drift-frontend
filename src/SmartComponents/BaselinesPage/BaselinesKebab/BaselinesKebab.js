import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import { Dropdown, KebabToggle, DropdownItem } from '@patternfly/react-core';
import DeleteBaselinesModal from '../DeleteBaselinesModal/DeleteBaselinesModal';

export class BaselinesKebab extends Component {
    constructor(props) {
        super(props);

        this.state = {
            kebabOpened: false,
            modalOpened: false
        };

        this.toggleKebab = () => {
            const { kebabOpened } = this.state;
            this.setState({
                kebabOpened: !kebabOpened
            });
        };

        this.toggleModalOpened = () => {
            const { modalOpened } = this.state;
            this.setState({
                modalOpened: !modalOpened,
                kebabOpened: false
            });
        };

        this.clearFilters = () => {
            const { clearFilters } = this.props;
            this.setState({
                kebabOpened: false
            });
            clearFilters();
        };
    }

    render() {
        const { kebabOpened, modalOpened } = this.state;
        const { clearFilters, fetchWithParams, selectedBaselineIds, tableId } = this.props;
        let dropdownItems;

        dropdownItems = [
            <DropdownItem
                key="multi-delete"
                component="button"
                onClick={ this.toggleModalOpened }
                isDisabled={ selectedBaselineIds.length < 1 }
            >
                Delete baselines
            </DropdownItem>,
            <DropdownItem
                key="clear-filters"
                component="button"
                onClick={ clearFilters ? this.clearFilters : null }
                isDisabled={ clearFilters ? false : true }
            >
                Clear filters
            </DropdownItem>
        ];

        return (
            <React.Fragment>
                { modalOpened
                    ? <DeleteBaselinesModal
                        modalOpened={ true }
                        tableId={ tableId }
                        fetchWithParams={ fetchWithParams }
                    />
                    : null
                }
                <Dropdown
                    style={ { float: 'left' } }
                    toggle={ <KebabToggle onToggle={ this.toggleKebab } /> }
                    isOpen={ kebabOpened }
                    dropdownItems={ dropdownItems }
                    isPlain
                />
            </React.Fragment>
        );
    }
}

BaselinesKebab.propTypes = {
    fetchWithParams: PropTypes.func,
    selectedBaselineIds: PropTypes.array,
    tableId: PropTypes.string,
    clearFilters: PropTypes.func
};

function mapStateToProps(state) {
    return {
        selectedBaselineIds: state.baselinesTableState.checkboxTable.selectedBaselineIds
    };
}

export default connect(mapStateToProps, null)(BaselinesKebab);
