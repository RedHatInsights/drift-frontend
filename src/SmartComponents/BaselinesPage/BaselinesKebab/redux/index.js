export { default as baselinesKebabActions } from './actions';
