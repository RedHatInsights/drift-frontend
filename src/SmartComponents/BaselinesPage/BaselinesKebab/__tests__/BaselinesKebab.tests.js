import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { BaselinesKebab } from '../BaselinesKebab';

describe('BaselinesKebab', () => {
    let props;

    beforeEach(() => {
        props = {
            selectedBaselineIds: []
        };
    });

    it('should render correctly', () =>{
        const wrapper = shallow(
            <BaselinesKebab { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render kebabOpened', () =>{
        const wrapper = shallow(
            <BaselinesKebab { ...props }/>
        );
        wrapper.instance().toggleKebab();
        expect(wrapper.state('kebabOpened')).toEqual(true);
    });

    it('should render modalOpened', () =>{
        const wrapper = shallow(
            <BaselinesKebab { ...props }/>
        );
        wrapper.instance().toggleModalOpened();
        expect(wrapper.state('modalOpened')).toEqual(true);
    });
});
