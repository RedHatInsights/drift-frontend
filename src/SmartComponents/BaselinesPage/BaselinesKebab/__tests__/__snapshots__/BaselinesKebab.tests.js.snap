// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaselinesKebab should render correctly 1`] = `
<Fragment>
  <Dropdown
    dropdownItems={
      Array [
        <DropdownItem
          component="button"
          isDisabled={true}
          onClick={[Function]}
        >
          Delete baselines
        </DropdownItem>,
        <DropdownItem
          component="button"
          isDisabled={true}
          onClick={null}
        >
          Clear filters
        </DropdownItem>,
      ]
    }
    isOpen={false}
    isPlain={true}
    style={
      Object {
        "float": "left",
      }
    }
    toggle={
      <KebabToggle
        onToggle={[Function]}
      />
    }
  />
</Fragment>
`;

exports[`ConnectedBaselinesKebab should render correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Connect(BaselinesKebab)>
        <BaselinesKebab
          dispatch={[Function]}
          selectedBaselineIds={Array []}
        >
          <Dropdown
            dropdownItems={
              Array [
                <DropdownItem
                  component="button"
                  isDisabled={true}
                  onClick={[Function]}
                >
                  Delete baselines
                </DropdownItem>,
                <DropdownItem
                  component="button"
                  isDisabled={true}
                  onClick={null}
                >
                  Clear filters
                </DropdownItem>,
              ]
            }
            isOpen={false}
            isPlain={true}
            style={
              Object {
                "float": "left",
              }
            }
            toggle={
              <KebabToggle
                onToggle={[Function]}
              />
            }
          >
            <DropdownWithContext
              autoFocus={true}
              className=""
              direction="down"
              dropdownItems={
                Array [
                  <DropdownItem
                    component="button"
                    isDisabled={true}
                    onClick={[Function]}
                  >
                    Delete baselines
                  </DropdownItem>,
                  <DropdownItem
                    component="button"
                    isDisabled={true}
                    onClick={null}
                  >
                    Clear filters
                  </DropdownItem>,
                ]
              }
              isGrouped={false}
              isOpen={false}
              isPlain={true}
              menuAppendTo="inline"
              onSelect={[Function]}
              position="left"
              style={
                Object {
                  "float": "left",
                }
              }
              toggle={
                <KebabToggle
                  onToggle={[Function]}
                />
              }
            >
              <div
                className="pf-c-dropdown"
                data-ouia-component-id={0}
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe={true}
                style={
                  Object {
                    "float": "left",
                  }
                }
              >
                <KebabToggle
                  aria-haspopup={true}
                  getMenuRef={[Function]}
                  id="pf-dropdown-toggle-id-0"
                  isOpen={false}
                  isPlain={true}
                  key=".0"
                  onEnter={[Function]}
                  onToggle={[Function]}
                  parentRef={
                    Object {
                      "current": <div
                        class="pf-c-dropdown"
                        data-ouia-component-id="0"
                        data-ouia-component-type="PF4/Dropdown"
                        data-ouia-safe="true"
                        style="float: left;"
                      >
                        <button
                          aria-expanded="false"
                          aria-haspopup="true"
                          aria-label="Actions"
                          class="pf-c-dropdown__toggle pf-m-plain"
                          id="pf-dropdown-toggle-id-0"
                          type="button"
                        >
                          <svg
                            aria-hidden="true"
                            fill="currentColor"
                            height="1em"
                            role="img"
                            style="vertical-align: -0.125em;"
                            viewBox="0 0 192 512"
                            width="1em"
                          >
                            <path
                              d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                              transform=""
                            />
                          </svg>
                        </button>
                      </div>,
                    }
                  }
                >
                  <Toggle
                    aria-haspopup={true}
                    aria-label="Actions"
                    bubbleEvent={false}
                    className=""
                    getMenuRef={[Function]}
                    id="pf-dropdown-toggle-id-0"
                    isActive={false}
                    isDisabled={false}
                    isOpen={false}
                    isPlain={true}
                    isPrimary={false}
                    isSplitButton={false}
                    onEnter={[Function]}
                    onToggle={[Function]}
                    parentRef={
                      Object {
                        "current": <div
                          class="pf-c-dropdown"
                          data-ouia-component-id="0"
                          data-ouia-component-type="PF4/Dropdown"
                          data-ouia-safe="true"
                          style="float: left;"
                        >
                          <button
                            aria-expanded="false"
                            aria-haspopup="true"
                            aria-label="Actions"
                            class="pf-c-dropdown__toggle pf-m-plain"
                            id="pf-dropdown-toggle-id-0"
                            type="button"
                          >
                            <svg
                              aria-hidden="true"
                              fill="currentColor"
                              height="1em"
                              role="img"
                              style="vertical-align: -0.125em;"
                              viewBox="0 0 192 512"
                              width="1em"
                            >
                              <path
                                d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                                transform=""
                              />
                            </svg>
                          </button>
                        </div>,
                      }
                    }
                  >
                    <button
                      aria-expanded={false}
                      aria-haspopup={true}
                      aria-label="Actions"
                      className="pf-c-dropdown__toggle pf-m-plain"
                      disabled={false}
                      id="pf-dropdown-toggle-id-0"
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      type="button"
                    >
                      <EllipsisVIcon
                        color="currentColor"
                        noVerticalAlign={false}
                        size="sm"
                      >
                        <svg
                          aria-hidden={true}
                          aria-labelledby={null}
                          fill="currentColor"
                          height="1em"
                          role="img"
                          style={
                            Object {
                              "verticalAlign": "-0.125em",
                            }
                          }
                          viewBox="0 0 192 512"
                          width="1em"
                        >
                          <path
                            d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                            transform=""
                          />
                        </svg>
                      </EllipsisVIcon>
                    </button>
                  </Toggle>
                </KebabToggle>
              </div>
            </DropdownWithContext>
          </Dropdown>
        </BaselinesKebab>
      </Connect(BaselinesKebab)>
    </Provider>
  </Router>
</MemoryRouter>
`;
